src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import java.util.function.Supplier;
+
+import static edu.wpi.first.units.Units.Radians;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE = null;
+    private final HoodInputsAutoLogged inputs = HoodIO.inputs;
+    private final HoodIO io;
+    private final Mechanism2d mechanism2d = new Mechanism2d(
+            3,3
+    );
+    private final MechanismRoot2d root = mechanism2d.getRoot("Hood", HoodConstants.mechanism2dPose.getFirst(), HoodConstants.mechanism2dPose.getSecond());
+    private final MechanismLigament2d hood = root.append(
+        new MechanismLigament2d("Hood", HoodConstants.HoodLength, 45)
+    );
+
+    private ControlMode controlMode = ControlMode.POSITION;
+
+    /**
+     * Constructor for Hood subsystem.
+     *
+     * @param io IO of the subsystem.
+     */
+    private Hood(HoodIO io) {
+        this.io = io;
+    }
+
+    /**
+     * Gets the single instance of hoodSubsystem.
+     *
+     * @return The single instance of hoodSubsystem.
+     */
+    public static Hood getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(HoodIO io) {
+        INSTANCE = new Hood(io);
+    }
+
+    public MutableMeasure<Angle> getAngle() {
+        return inputs.angle;
+    }
+
+    public boolean atSetpoint() {
+        return io.atSetpoint();
+    }
+
+    public Command setAngle(Supplier<Rotation2d> angle){
+        controlMode = ControlMode.POSITION;
+        return runOnce(() -> inputs.angleSetpoint.mut_replace(angle.get().getRadians(), Radians));
+    }
+
+    public Command setPower(Supplier<Double> power){
+        controlMode = ControlMode.POWER;
+        return runOnce(() -> inputs.powerSetpoint = power.get());
+    }
+
+    public Command getResetAbsoluteEncoderCommand(){
+        setPower(() -> -0.3);
+        return runOnce(() -> io.resetAbsoluteEncoder());
```

Comment:
This is an incorrect use of commands. The setPower here wouldn't actually happen. Instead, you have to chain the commands as such:
```suggestion
        return setPower(() -> -0.3).andThen(runOnce(() -> io.resetAbsoluteEncoder()));
```
Also, -0.3 is a magic number that should be in the constants.